File: HandloomAPI/Models/Sales/RemoveStockModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace HandloomAPI.Models.Sales
{
    public class RemoveStock_ViewModel
    {
        public long RemoveStockID { get; set; }
        public string RemoveStockNo { get; set; }
        public DateTime RemoveStockDate { get; set; }
        public long ModifiedBy { get; set; }
        public DateTime ModifiedDate { get; set; }
        public string ModifiedByName { get; set; }
        public string Remarks { get; set; }
        public decimal TotalQty { get; set; }
        public bool IsLocked { get; set; }
        public long BrandID { get; set; }
        public string BrandName { get; set; }
    }

    public class RemoveStock_AddModel
    {
        public long RemoveStockID { get; set; }
        public string RemoveStockNo { get; set; }
        public DateTime RemoveStockDate { get; set; }
        public long BrandID { get; set; }
        public string Remarks { get; set; }
        public List<RemoveStock_Item_AddModel> Items { get; set; }
        public decimal TotalQty { get; set; }
    }
    public class RemoveStock_Item_AddModel
    {
        public long ID { get; set; }
        public long SrNo { get; set; }
        public long StockItemID { get; set; }
        public long ParentProductID { get; set; }
        public long ProductID { get; set; }
        public List<RemoveStock_Item_Attribute_AddModel> AttributeValues { get; set; }
        public decimal Qty { get; set; }
        public long HSNCodeID { get; set; }
        public string Remarks { get; set; }
    }
    public class RemoveStock_Item_Attribute_AddModel
    {
        public long ProductAttributeID { get; set; }
        public long ProductAttributeValueID { get; set; }
    }
    public class RemoveStock_Item_ViewModel
    {
        public long ID { get; set; }
        public long RemoveStockID { get; set; }
        public long SrNo { get; set; }
        public long StockItemID { get; set; }   
        public long ParentProductID { get; set; }   
        public long ProductID { get; set; }
        public string sku { get; set; }
        public string ProductName { get; set; }
        public decimal Qty { get; set; }
        public long HSNCodeID { get; set; }
        public string HSNCodeName { get; set; }
        public string Remarks { get; set; }
        public string Photo { get; set; }
        public List<RemoveStock_Item_Attribute_ViewModel> AttributeValues { get; set; }
    }
    public class RemoveStock_Parent_Item_ViewModel
    {
        public long RemoveStockID { get; set; }
        public long SrNo { get; set; }
        public long ProductID { get; set; }
        public string sku { get; set; }
        public string ProductName { get; set; }
        public decimal Qty { get; set; }
        public string Photo { get; set; }
    }
    public class RemoveStock_Item_Attribute_ViewModel
    {
        public long RemoveStockID { get; set; }
        public long RemoveStockItemID { get; set; }
        public long ProductAttributeID { get; set; }
        public long ProductAttributeValueID { get; set; }
        public string name { get; set; }
        public string option { get; set; }
    }
    public class RemoveStock_Filter
    {
        public long RemoveStockID { get; set; }
        public DateTime? FromDate { get; set; }
        public DateTime? ToDate { get; set; }
        public string RemoveStockNo { get; set; }
        public long PartyID { get; set; }
        public long BrandID { get; set; }
    }
}

File: TradebullsAngular/src/app/Models/Sales/RemoveStockModel.ts
export class RemoveStock_AddModel {
    RemoveStockID?:number;
    RemoveStockNo?:string;
    RemoveStockDate?:Date;
    BrandID?:number;
    Remarks?:string;
    TotalQty?: number;
    Items:RemoveStock_Item_AddModel[] = [];
}

export class RemoveStock_Item_AddModel {
    ID?:number;
    SrNo?:number;
    StockItemID?:number;
    ParentProductID?:number;
    ProductID?:number;
    AttributeValues:{ProductAttributeID: number, ProductAttributeValueID: number} [] = [];
    Qty?:number;
    HSNCodeID?:number;
    Remarks?:string;
}

export class RemoveStock_Item{
    ID: number = 0;
    SrNo: number = 0;
    StockItemID: number = 0;
    ParentProductID: number = 0;
    ProductID: number = 0;
    sku?: string;
    ProductName?: string;
    AttributeValues: {
      ProductAttributeID: number,
      ProductAttributeValueID: number,
      Name: string,
      Option: string
    }[] = [];
    Qty: number = 0;
    HSNCodeID?: number;
    HSNCodeName?: string;
    Remarks?: string;
    Photo?: string;
}